lib/bootloader_setup.pm

```diff
@@ -64,120 +78,154 @@ sub select_bootmenu_option {
         # and UEFI menues.
         send_key 'up';
     }
-    if (get_var('ZDUP') || get_var("ONLINE_MIGRATION")) {
-        send_key 'ret';    # boot from hard disk
+    if (get_var('ZDUP') || get_var('ONLINE_MIGRATION')) {
+        boot_local_disk;
         return 3;
     }
 
-    if (get_var("UPGRADE")) {
+    if (get_var('UPGRADE')) {
         # random magic numbers
-        send_key_until_needlematch('inst-onupgrade', 'down', 10, 5);
+        if (check_var('ARCH', 'ppc64le')) {
+            send_key_until_needlematch 'inst-onupgrade', 'up', 10, 5;
```

Comment:
why?

Also, if you like you could write

```perl
send_key_until_needlematch 'inst-onupgrade', check_var('ARCH', 'ppc64le') ? 'up' : 'down', 10, 5;
```

or maybe better

```perl
# <here comes the explanation why ppc64le needs to be special>
my $direction = check_var('ARCH', 'ppc64le') ? 'up' : 'down';
send_key_until_needlematch 'inst-onupgrade', $direction, 10, 5;
```

because you also need the direction further down again